command/agent/local.go

```diff
@@ -379,6 +379,12 @@ func (l *localState) setSyncState() error {
 		}
 
 		// If our definition is different, we need to update it
+		if existing.EnableTagDrift {
+			l.logger.Printf("[DEBUG] Tag drift enabled.")
+			existing.Tags = service.Tags
```

Comment:
@ryanuber That makes sense, but I believe it's a little out of scope for this particular pull request.  I would be happy to contribute independent of this change.